Sources/SwiftSyntaxBuilderGeneration/Tokens.swift.gyb

```diff
@@ -0,0 +1,135 @@
+  }
+}
+
+public class SwiftKeyword: Keyword { }
+
+public class DeclKeyword: SwiftKeyword { }
+
+public class StmtKeyword: SwiftKeyword { }
+
+public class ExprKeyword: SwiftKeyword { }
+
+public class PatternKeyword: SwiftKeyword { }
+
+public class SilKeyword: Keyword { }
+
+public class PoundKeyword: Token {
+  public init(name: String, kind: String, serializationCode: Int, text: String, classification: String = "Keyword") {
+    super.init(name: name, kind: "pound_\(kind)", serializationCode: serializationCode, unprefixedKind: kind, text: text, classification: classification, isKeyword: true, requiresTrailingSpace: true)
+  }
+}
+
+public class PoundObjectLiteral: PoundKeyword {
+  public let description: String
+  public let `protocol`: String
+
+  public init(name: String, kind: String, serializationCode: Int, text: String, classification: String = "ObjectLiteral", description: String, `protocol`: String) {
+    self.description = `description`
+    self.`protocol` = `protocol`
+    super.init(name: name, kind: kind, serializationCode: serializationCode, text: text, classification: classification)
+  }
+}
+
+public class PoundConfig: PoundKeyword { }
+
+public class PoundDirectiveKeyword: PoundKeyword {
+  public override init(name: String, kind: String, serializationCode: Int, text: String, classification: String = "PoundDirectiveKeyword") {
+    super.init(name: name, kind: kind, serializationCode: serializationCode, text: text, classification: classification)
+  }
+}
+
+public class PoundConditionalDirectiveKeyword: PoundDirectiveKeyword {
+  public override init(name: String, kind: String, serializationCode: Int, text: String, classification: String = "PoundDirectiveKeyword") {
+    super.init(name: name, kind: kind, serializationCode: serializationCode, text: text, classification: classification)
+  }
+}
+
+public class Punctuator: Token { }
+
+public class Literal: Token { }
+
+public class Misc: Token { }
+
+let SYNTAX_TOKENS: [Token] = [
+% for token in SYNTAX_TOKENS:
+%   class_name = type(token).__name__
+%   text = 'text: "' + token.text + '", ' if token.text else ""
+%   if class_name in ['Keyword', 'SwiftKeyword', 'DeclKeyword', 'StmtKeyword', 'ExprKeyword', 'PatternKeyword', 'SilKeyword']:
+    ${class_name}(name: "${token.name}", serializationCode: ${token.serialization_code}, text: "${token.text}"),
+%   elif class_name in ['PoundKeyword', 'PoundConfig', 'PoundDirectiveKeyword', 'PoundConditionalDirectiveKeyword']:
+    ${class_name}(name: "${token.name}", kind: "${token.kind}", serializationCode: ${token.serialization_code}, text: "${token.text}"),
+%   elif class_name in ['Punctuator', 'Misc']:
+    ${class_name}(name: "${token.name}", kind: "${token.kind}", serializationCode: ${token.serialization_code}, ${text}requiresLeadingSpace: ${str(token.requires_leading_space).lower()}, requiresTrailingSpace: ${str(token.requires_trailing_space).lower()}),
```

Comment:
Small nitpick: If we omitted `requiresLeadingSpace` and `requiresTrailingSpace` if their value is `False`, I think the generated code would pretty much 1:1 match what we have in Token.py, which I think would be kind of nice. It’s not really required but would be a nice touch, I think.

Reply:
Will fix this in a follow up PR if this gets merged